libraries/lib-wave-track/AudioSegmentSampleView.h

```diff
@@ -0,0 +1,65 @@
+/*  SPDX-License-Identifier: GPL-2.0-or-later */
+/*!********************************************************************
+
+  Audacity: A Digital Audio Editor
+
+  AudioSegmentSampleView.h
+
+  Matthieu Hodgkinson
+
+**********************************************************************/
+#pragma once
+
+#include "SampleCount.h"
+
+#include <chrono>
+#include <fstream>
+#include <memory>
+#include <vector>
+
+using BlockSampleView = std::shared_ptr<std::vector<float>>;
+
+//! An audio segment is either a whole clip or the silence between clips.
+class WAVE_TRACK_API AudioSegmentSampleView final
+{
+public:
+   /*
+    * @brief Constructs an `AudioSegmentSampleView` from `BlockSampleView`s.
+    * @param blockViews A continuous sequence of samples, segmented in blocks.
+    * @param start Index of the first sample in `blockViews` to look at.
+    * @param length The number of samples from `start` to look at.
+    * @pre `!blockViews.empty()`
+    * @pre `start < blockViews[0].size()`
+    * @pre `length < sum(sizes(blockViews)) - start`
```

Comment:
The last `@pre` is not quite executable as written, so if I wrote it, I would not put only variable names in grave accents and leave the statement informal.